api/queries_repo.go

```diff
@@ -59,6 +60,46 @@ func (r Repository) ViewerCanPush() bool {
 	}
 }
 
+func RepoExistsOnGitHub(client *Client, repo ghrepo.Interface) (bool, error) {
```

Comment:
Nit: how about `GitHubRepoExists` as a name, to make it sound like a variant of `GitHubRepo` (which it is)?